framework/packages/abstract-testing/src/abstract_mock_querier.rs

```diff
@@ -6,103 +6,68 @@ use abstract_std::{
             // Setup the account owner as the test owner
             .with_contract_item(
                 account_base.addr(),
                 Item::new(OWNERSHIP_STORAGE_KEY),
                 &Some(Ownership {
                     owner: GovernanceDetails::Monarchy {
-                        monarch: self.abstract_addrs.owner.clone(),
+                        monarch: abstract_addrs.owner.clone(),
                     },
                     pending_owner: None,
                     pending_expiry: None,
                 }),
             )
             .with_contract_map_entry(
-                &self.abstract_addrs.version_control,
+                &abstract_addrs.version_control,
                 ACCOUNT_ADDRESSES,
                 (&account_id, account_base.clone()),
-            );
-
-        self
-    }
-    // ANCHOR_END: account
-
-    /// Add mock assets into ANS
-    pub fn assets(mut self, assets: Vec<(&AssetEntry, AssetInfo)>) -> Self {
-        self.builder = self.builder.with_contract_map_entries(
-            &self.abstract_addrs.ans_host,
-            ASSET_ADDRESSES,
-            assets,
-        );
-
-        self
-    }
-
-    pub fn contracts(mut self, contracts: Vec<(&ContractEntry, Addr)>) -> Self {
-        self.builder = self.builder.with_contract_map_entries(
-            &self.abstract_addrs.ans_host,
-            CONTRACT_ADDRESSES,
-            contracts,
-        );
-
-        self
+            )
     }
 
-    pub fn channels(mut self, channels: Vec<(&ChannelEntry, String)>) -> Self {
-        self.builder = self.builder.with_contract_map_entries(
-            &self.abstract_addrs.ans_host,
-            CHANNELS,
-            channels,
-        );
-
-        self
+    fn assets(self, assets: Vec<(&AssetEntry, AssetInfo)>) -> Self {
+        let abstract_addrs = self.addrs();
+        self.with_contract_map_entries(&abstract_addrs.ans_host, ASSET_ADDRESSES, assets)
     }
 
-    /// Change the version control address. Any account added after this will be registered to this address.
-    pub fn set_version_control(mut self, version_control: Addr) -> Self {
-        self.abstract_addrs.version_control = version_control;
-        self
-    }
+    fn contracts(self, contracts: Vec<(&ContractEntry, Addr)>) -> Self {
+        let abstract_addrs = self.addrs();
 
-    pub fn ans(self, _ans: MockAnsHost) -> Self {
-        self
+        self.with_contract_map_entries(&abstract_addrs.ans_host, CONTRACT_ADDRESSES, contracts)
     }
 
-    pub fn builder(self) -> MockQuerierBuilder {
-        self.builder
-    }
+    fn channels(self, channels: Vec<(&ChannelEntry, String)>) -> Self {
+        let abstract_addrs = self.addrs();
 
-    pub fn addrs(&self) -> &AbstractMockAddrs {
-        &self.abstract_addrs
+        self.with_contract_map_entries(&abstract_addrs.ans_host, CHANNELS, channels)
     }
 
-    pub fn build(self) -> MockQuerier {
-        self.builder.build()
+    fn addrs(&self) -> AbstractMockAddrs {
```

Comment:
How valuable do you think it is that this fn doesn't require API to be passed?

Reply:
Other methods of this trait also need it